syntax = "proto3";

option csharp_namespace = "GrpcServer";
package proto;

import "google/protobuf/timestamp.proto";

service Storage {
  rpc Read(ReadRequest) returns (ReadResponse);
  rpc Write(WriteRequest) returns (WriteResponse);
}

message WriteRequest {
  string FileName = 1;
  string Value = 2;
  google.protobuf.Timestamp Timestamp = 3;
}

message ReadRequest { string FileName = 1;}

message WriteResponse {
  int32 Ok = 1;
}

message ReadResponse{
  string Value = 1; 
  int32 Ok = 2; 
  google.protobuf.Timestamp Timestamp = 3;
}